cmd/nknd/nknd.go

```diff
@@ -197,12 +199,46 @@ func nknMain(c *cli.Context) error {
 		}
 	}
 
+	if config.Parameters.Hostname == "" { // Skip query self extIP via set "HostName" in config.json
+		log.Info("Getting my IP address...")
+		var extIP string
+		if createMode { // There is no seed available in create mode, used ipify
+			extIP, err = ipify.GetIp()
+		} else {
+			extIP, err = AskMyIP(config.Parameters.SeedList)
+		}
+		if err != nil {
+			return err
+		}
+		log.Infof("My IP address is %s", extIP)
+		config.Parameters.Hostname = "64.64.245.149"
+	}
+
+	wssCertExists, httpsCertExists, prePreparedCert, _ := certs.CertExists()
+	wssCertReady := make(chan struct{}, 1)
+	httpsCertReady := make(chan struct{}, 1)
+
+	if wssCertExists {
+		go func() {
+			wssCertReady <- struct{}{}
+		}()
+	}
+	if httpsCertExists {
+		go func() {
+			httpsCertReady <- struct{}{}
+		}()
+	}
+	if !(wssCertExists && httpsCertExists) || !prePreparedCert {
+		go certs.PrepareCert(wssCertExists, httpsCertExists, wssCertReady, httpsCertReady)
+	}
+
 	// init web service
 	dashboard.Init(nil, wallet, nil)
 
 	// start JsonRPC
 	rpcServer := httpjson.NewServer(nil, wallet)
-	rpcServer.Start()
+	rpcServer.Init()
```

Comment:
Suggest to move Init, StartHttp, StartTls into a single function call Start() just like websocket part